packages/server/src/course/ical.service.ts

```diff
@@ -190,6 +190,10 @@ export class IcalService {
     let queue = await QueueModel.findOne({
       where: { courseId: course.id, room: 'Online' },
     });
+    const semester =
+      (await SemesterModel.findOne({
+        where: { id: course.semesterId },
+      })) || course.semester;
```

Comment:
is `course.semester` supposed to go first in the `OR` here? this way we dont run the db query if we already have the sem.

Reply:
yes i can fix that -- in prod course.semester is null its just the factories that dont give a semesterid but give a semestermodel object